RecoTracker/LSTCore/interface/alpaka/Constants.h

```diff
@@ -0,0 +1,183 @@
+#ifndef RecoTracker_LSTCore_interface_alpaka_Constants_h
+#define RecoTracker_LSTCore_interface_alpaka_Constants_h
+
+#include <alpaka/alpaka.hpp>
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+
+#ifdef CACHE_ALLOC
+#include "HeterogeneousCore/AlpakaInterface/interface/CachedBufAlloc.h"
+#endif
+
+#ifdef ALPAKA_ACC_GPU_CUDA_ENABLED
+#include <cuda_fp16.h>
+#endif
+
+namespace SDL {
+// Half precision wrapper functions.
+#if defined(FP16_Base)
+#define __F2H __float2half
+#define __H2F __half2float
+  typedef __half float FPX;
+#else
+#define __F2H
+#define __H2F
+  typedef float FPX;
+#endif
+
+  using Idx = alpaka_common::Idx;
+  using Dim = alpaka_common::Dim3D;
+  using Dim1d = alpaka_common::Dim1D;
+  using Vec = alpaka_common::Vec3D;
+  using Vec1d = alpaka_common::Vec1D;
+  using WorkDiv = alpaka_common::WorkDiv3D;
+
+  using Acc = ALPAKA_ACCELERATOR_NAMESPACE::Acc3D;
+  using Dev = ALPAKA_ACCELERATOR_NAMESPACE::Device;
+  using DevHost = ALPAKA_ACCELERATOR_NAMESPACE::DevHost;
+  using QueueAcc = ALPAKA_ACCELERATOR_NAMESPACE::Queue;
+
+  Vec const elementsPerThread(Vec::all(static_cast<Idx>(1)));
+
+// Needed for files that are compiled by g++ to not throw an error.
+// uint4 is defined only for CUDA, so we will have to revisit this soon when running on other backends.
+#if !defined(ALPAKA_ACC_GPU_CUDA_ENABLED) && !defined(ALPAKA_ACC_GPU_HIP_ENABLED)
+  struct uint4 {
+    unsigned int x;
+    unsigned int y;
+    unsigned int z;
+    unsigned int w;
+  };
+#endif
+
+  // Buffer type for allocations where auto type can't be used.
+  template <typename TDev, typename TData>
+  using Buf = alpaka::Buf<TDev, TData, Dim1d, Idx>;
+
+  // Allocation wrapper function to make integration of the caching allocator easier and reduce code boilerplate.
+  template <typename T, typename TAcc, typename TSize, typename TQueue>
+  ALPAKA_FN_HOST ALPAKA_FN_INLINE Buf<alpaka::Dev<TAcc>, T> allocBufWrapper(TAcc const& devAccIn,
+                                                                            TSize nElements,
+                                                                            TQueue queue) {
+#ifdef CACHE_ALLOC
+    return cms::alpakatools::allocCachedBuf<T, Idx>(devAccIn, queue, Vec1d(static_cast<Idx>(nElements)));
+#else
+    return alpaka::allocBuf<T, Idx>(devAccIn, Vec1d(static_cast<Idx>(nElements)));
+#endif
+  }
+
+  // Second allocation wrapper function when queue is not given. Reduces code boilerplate.
+  template <typename T, typename TAcc, typename TSize>
+  ALPAKA_FN_HOST ALPAKA_FN_INLINE Buf<alpaka::Dev<TAcc>, T> allocBufWrapper(TAcc const& devAccIn, TSize nElements) {
+    return alpaka::allocBuf<T, Idx>(devAccIn, Vec1d(static_cast<Idx>(nElements)));
+  }
+
+  // Wrapper function to reduce code boilerplate for defining grid/block sizes.
+  ALPAKA_FN_HOST ALPAKA_FN_INLINE Vec createVec(int x, int y, int z) {
+    return Vec(static_cast<Idx>(x), static_cast<Idx>(y), static_cast<Idx>(z));
+  }
```

Comment:
You can just use
```c++
  Vec const threadsPerBlock1{1u, 1u, 256u};
```
instead of
```c++
  Vec const threadsPerBlock1 = createVec(1, 1, 256);
```

If you prefer to keep the helper function, I would suggest renaming the arguments to
```suggestion
  // Wrapper function to reduce code boilerplate for defining grid/block sizes.
  ALPAKA_FN_HOST ALPAKA_FN_INLINE Vec createVec(int z, int y, int x) {
    return Vec(static_cast<Idx>(z), static_cast<Idx>(y), static_cast<Idx>(x));
  }
```
to avoid confusion with the CUDA nomenclature.